### DISCUSSION TITLE
Error while creating a copilot from a snapshot

### DISCUSSION THREAD

**Discussion by aakashnandi4** [microsoft/powerplatform-build-tools#1042] [Q&A]
While working on an incident I tried recreating a bot using the bot-snaphot share by the customer. 
The error message I see is the following : 

Error: HTTP Status:BadRequest, ServiceRequestId: 7d3abb3f-9a6a-469e-8fcd-19f1412d9e0b [0x80048408:InvalidStateCodeStatusCode] 2 is not a valid status code for state code botcomponentState.Active on botcomponent with Id 957d2dd3-6a1d-4a9b-b95a-1b5d8cc07d0d.

Upon searching for the botComponent ID in the error message , in the bot-snapshot, I could not find the component that caused this error. 

It would be helpful if the error message spits out either schemaName or displayName of the component.

**Comment by jake-mannion-std** [comment]
The _botcomponent_ table in the environment (_Copilot components_ is the display name) should enable you to identify the component associated with that guid.

Or just drop all inactive topics from the bot. If you are working with an exported yaml definition (pac cli) you can remove them there too, or flip them back to active. If removing, make sure to delete the entire DialogCompnent definition.

![image](https://github.com/user-attachments/assets/d6d48621-ca6b-4968-a189-92a16da94295)

Hope that helps. Had a similar hurdle today and web search led me here.
